sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -510,39 +526,44 @@ def __init__(self, **kwargs: Any) -> None:
         self._update_lock = Lock()
         self._refresh_lock = Lock()
 
-    def refresh(self, **kwargs) -> None:
+    def refresh(self, **kwargs) -> None:  # pylint: disable=too-many-statements
         if not self._refresh_on and not self._feature_flag_refresh_enabled:
-            logging.debug("Refresh called but no refresh enabled.")
+            logger.debug("Refresh called but no refresh enabled.")
             return
         if not self._refresh_timer.needs_refresh():
-            logging.debug("Refresh called but refresh interval not elapsed.")
+            logger.debug("Refresh called but refresh interval not elapsed.")
             return
         if not self._refresh_lock.acquire(blocking=False):  # pylint: disable= consider-using-with
-            logging.debug("Refresh called but refresh already in progress.")
+            logger.debug("Refresh called but refresh already in progress.")
             return
         success = False
         need_refresh = False
         error_message = """
                         Failed to refresh configuration settings from Azure App Configuration.
                         """
         exception: Exception = RuntimeError(error_message)
+        is_failover_request = False
+        client_count = self._replica_client_manager.get_client_count()
```

Comment:
This is tracking the number of replicas configured by user. We dont track auto discovered replicas in this metric. We can rename it to `replica_count` and also subtract 1 here instead of doing it inside the while loop.

Reply:
fixed

Reply:
We don't count auto discovered replicas? In all but .Net and Spring we will only have other auto discovered replicas.

Reply:
That's what we have in .NET today: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/90835cf5efbfb26d6de6068e4c31c3050947b78a/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L969

Auto discovered replicas are not included in the count. But I think we should include it. @amerjusupovic could you add a work item to track it for .NET provider?

Reply:
Yea looks like we just missed it in the replica discovery PR, I can add an issue tracking that.